### PR TITLE
refactor(api): shares and likes

### PR DESCRIPTION
**Pull request by radurogojanumai**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -124,8 +124,29 @@ mostly.share(g | sd | c, email)
 mostly.unshare(g | sd | c, email)
 
 g.shares()
+g.share(email)
```
Comment (mplatzer) [inline README.md:127]: let's just have a single line to demo syntax, i.e.
`g.share(email, "VIEW" | "EDIT" | "ADMIN")`

and drop the other 3 g.share() examples.

plus lets drop `mostly.share` and `mostly.unshare` from docs and from interface. this is the old deprecated interface.

File: README.md [hunk 2]
```diff
@@ -124,8 +124,29 @@ mostly.share(g | sd | c, email)
...
+g.share(email, "VIEW" | "view" | ""EDIT" | "edit" | "ADMIN" | "admin")
+g.share(email, PermissionLevel.view | PermissionLevel.edit | PermissionLevel.admin)
+g.unshare(email)
+g.share_public()
+g.unshare_public()
+g.like()
+g.unlike()
+
 sd.shares()
+sd.share(email)
```
Comment (mplatzer) [inline README.md:137]: `sd.share(email, "VIEW" | "EDIT" | "ADMIN")`

File: README.md [hunk 3]
```diff
@@ -124,8 +124,29 @@ mostly.share(g | sd | c, email)
...
+sd.unshare(email)
+sd.share_public()
+sd.unshare_public()
+sd.like()
+sd.unlike()
+
 c.shares()
+c.share(email)
+c.share(email, "EDIT")
```
Comment (mplatzer) [inline README.md:146]: `c.share(email, "VIEW" | "EDIT" | "ADMIN")`

and drop `c.share(email)`

File: tools/model.py [hunk 4]
```diff
@@ -257,7 +351,54 @@ def data(
...
+        """
+        self.client._share(
+            resource_id=self.id,
+            user_email=user_email,
+            permission_level=permission_level,
+        )
+
+    def unshare(self, user_email: str):
+        """
+        Stop sharing the Synthetic Data with the specified user
```
Comment (mplatzer) [inline tools/model.py:375]: use `synthetic dataset` consistently instead of `Synthetic Data` throughout the code

File: mostlyai/model.py [hunk 5]
```diff
@@ -353,7 +353,54 @@ def config(self) -> dict[str, Any]:
         return self.client._config(connector_id=self.id)
 
     def shares(self):
-        return self.client._shares(resource=self)
+        """
+        Retrieve list of shares
+
+        :return: list of ResourceShares
```
Comment (michdr) [inline mostlyai/model.py:359]: Just `ResourceShares`, which contains a list inside of that object.

Comment (michdr) [reply mostlyai/model.py:359]: Same goes with the other identical method signatures

**Review by michdr — APPROVED** [review]
A couple of minor corrections in docstrings. Other than that, looks good 👍🏼

File: README.md (hunk 1 above)
Comment (mplatzer) [reply README.md:127]: and please remove the `share_public` and `unshare_public` from the README, and comment out in the code. This is atm just for Admins, and we will want to rename that also in the Public API.

File: README.md [hunk 6]
```diff
@@ -117,15 +117,24 @@ sp = mostly.probe(g, size=size)
...
 ```
 
-### Sharing
+### Share and Like
 
 ```python
-mostly.share(g | sd | c, email)
-mostly.unshare(g | sd | c, email)
+c.shares()
+c.share(email, "VIEW" | ""EDIT" | "ADMIN")
```
Comment (zachtil) [inline README.md:124]: Duplicate double-quote

```suggestion
c.share(email, "VIEW" | "EDIT" | "ADMIN")
```

File: README.md [hunk 7]
```diff
@@ -117,15 +117,24 @@ sp = mostly.probe(g, size=size)
...
+c.unshare(email)
 
 g.shares()
+g.share(email, "VIEW" | ""EDIT" | "ADMIN")
```
Comment (zachtil) [inline README.md:128]: Duplicate double-quote

```suggestion
g.share(email, "VIEW" | "EDIT" | "ADMIN")
```

File: README.md [hunk 8]
```diff
@@ -117,15 +117,24 @@ sp = mostly.probe(g, size=size)
...
+g.unshare(email)
+g.like()
+g.unlike()
+
 sd.shares()
-c.shares()
+sd.share(email, "VIEW" | ""EDIT" | "ADMIN")
```
Comment (zachtil) [inline README.md:134]: Duplicate double-quote

```suggestion
sd.share(email, "VIEW" | "EDIT" | "ADMIN")
```

**Pull request closed without merging**